src/fp4_linear.py

```diff
@@ -0,0 +1,170 @@
+import torch
+import torch.nn as nn
+from typing import Optional
+
+try:
+    from flashinfer import nvfp4_quantize, mm_fp4, SfLayout
```

Comment:
I underspecified this, my apologies. For further work after this PR - specifically integrating this into Windows, do you recommend we just use cutlass kernel directly? Any windows-compatible library you might recommend which would provide similar interfaces?